src/module.c

```diff
@@ -6272,6 +6272,14 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
         serverAssert(ctx->module);
         reply = callReplyCreatePromise(ctx->module);
         c->bstate.async_rm_call_handle = reply;
+        if (!(call_flags & CMD_CALL_PROPAGATE_AOF)) {
+            /* No need for AOF propagation, set the relevant flags of the client */
+            c->flags |= CLIENT_MODULE_PREVENT_AOF_PROP;
```

Comment:
@oranagra , on `moduleReleaseTempClient` we are reseting the client flags and only keep `CLIENT_MODULE`:
```
c->flags = CLIENT_MODULE;
```
https://github.com/redis/redis/blob/cd58af4d7fe4b506899591b2846cfba61335c958/src/module.c#L618

Reply:
What I do not notice now is that maybe we want to have a MAX capacity to `moduleTempClients` today we do not have it. I guess its because we assume it can not grow to much (only on nested RM_Call), but now with blocking RM_Call it has a potential to grow? WDYT?

Reply:
i don't think we want to limit it...  we do have a mechanism to limit the cache size or actually to release it gradually when unused, i think that's sufficient, and if module or app will abuse it, we better let the resource utilization grow (assuming there are resources on the machine), rather than block something that could succeed.